src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/resource/CourtMaintenanceResource.kt

```diff
@@ -0,0 +1,111 @@
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = CourtDto::class))]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Information request to update court",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to make court update"
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Court ID not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+  @PutMapping("/id/{courtId}")
+  fun updateCourt(
+    @Schema(description = "Court ID", example = "ACCRYC", required = true)
+    @PathVariable @Size(max = 12, min = 2, message = "Court ID must be between 2 and 12") courtId: String,
+    @RequestBody courtUpdateRecord: UpdateCourtDto
+  ): CourtDto =
+    courtService.updateCourt(courtId, courtUpdateRecord)
+
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_REF_DATA')")
+  @PostMapping("")
+  @Operation(
+    summary = "Add a new court",
+    description = "Adds a new court information, role required is MAINTAIN_REF_DATA",
+    security = [ SecurityRequirement(name = "MAINTAIN_REF_DATA", scopes = ["write"])],
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(content = [Content(mediaType = "application/json", schema = Schema(implementation = CourtDto::class))]),
+    responses = [
+      ApiResponse(
+        responseCode = "201",
+        description = "Court Information Inserted",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = CourtDto::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Information request to add a court",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to make court insert"
+      )
+    ]
+  )
+  fun insertCourt(
+    @RequestBody courtInsertRecord: CourtDto
+  ): CourtDto =
+    courtService.insertCourt(courtInsertRecord)
+}
+
+@JsonInclude(NON_NULL)
+@Schema(description = "Court Update Record")
+data class UpdateCourtDto(
+  @Schema(description = "Name of the court", example = "Accrington Youth Court", required = true) @Size(max = 80, min = 2, message = "Court name must be between 2 and 80") @NotBlank val courtName: String,
+  @Schema(description = "Description of the court", example = "Accrington Youth Court", required = false) @Size(max = 200, min = 2, message = "Court name must be between 2 and 200") val courtDescription: String?,
```

Comment:
2 and 200 characters?

Reply:
Yes more of a placeholder not sure what min should be?